cmd/queue/main.go

```diff
@@ -152,47 +149,61 @@ func probeUserContainer() bool {
 	return err == nil
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	proxy := proxyForRequest(r)
+// Make handler a closure for testing.
+func handler(rc chan queue.ReqEvent, b *queue.Breaker, h1, h2 *httputil.ReverseProxy) func(http.ResponseWriter, *http.Request) {
+	reqChan := rc
```

Comment:
why do you need these variables?
If for naming, then just name the arguments this way.

Reply:
You're right, using the params directly works. :)